test/classes/initializers/base-class-generic-args.chpl

```diff
@@ -0,0 +1,44 @@
+// Modified from
+// test/classes/constructors/base-class-generic-args.chpl
+
+// Contributed by Rafael Larrosa Jimenez (rlarrosa@uma.es)
+
+class Circle {
+  param rank: int = 1;
+  type idxType = int;
+  const r : rank*idxType;
+  proc init(h,param rank=h.size)
+  {
+    this.r=h;
```

Comment:
I think this code should work as written, but it's not what I thought you were initially implementing. In particular, if Phase 2 is the default, and you can't update a const field in Phase 2, this would not be a valid initializer. I personally think we should just adjust the design to allow this initializer, but I wanted to call attention to the issue in case you missed it.